### PR TITLE
Randchat settings based on aff

### PR DESCRIPTION
**Pull request by multimokia**
Makes randchat naturally slow down as your affection gets lower.

This also changes the default randchat setting to be slower (what was `Less often` is now `Normal`), and the other options have been renamed.
As well, repeat topics now *starts* enabled.

The following table shows the new names for the levels
| Old | New |
|:-:|:-:|
| Often | Very Often |
| Normal | Often |
| Less Often | Normal |
| Occasionally | Less Often |
| Rarely | Occasionally |
| Very Rarely | Rarely |

Additionally, their values have been reversed so the slider works in the correct direction (left being lower, right being higher)

The levels available according to affection are as follows:
| Aff | Available Settings |
|:-:|:-:|
| Broken | Never- Rarely |
| Distressed | Never - Occasionally |
| Upset | Never - Less Often |
| Normal | Never - Normal |
| Happy | Never - Normal |
| Affectionate | Never - Often |
| Enamored | Never - Very Often |
| Love | Never - Very Often |

# Testing:
- Verify no crashes
- Verify randchat is decreased to the level max for the affection level if it drops below the old level's max
- Verify randchat slider gets less options as affection drops
- Verify randchat slider gets more options as affection is gained
- Verify no text clipping on menu
- Verify we default at `Normal`
- Verify we default repeat topics to `True`

### CODE REVIEW

File: Monika After Story/game/script-affection.rpy [hunk 1]
```diff
@@ -451,6 +461,10 @@ init 15 python in mas_affection:
         layout.QUIT_NO = mas_layout.QUIT_NO_BROKEN
         layout.QUIT = mas_layout.QUIT_BROKEN
 
+        #Change randchat
+        if store.persistent._mas_randchat_freq <= store.mas_randchat.OCCASIONALLY:
```
Comment (Booplicate) [inline Monika After Story/game/script-affection.rpy:465]: This should be the other way around, shouldn't it?
```suggestion
        if store.persistent._mas_randchat_freq >= store.mas_randchat.OCCASIONALLY:
```

File: Monika After Story/game/script-affection.rpy [hunk 2]
```diff
@@ -476,6 +490,10 @@ init 15 python in mas_affection:
             renpy.store.monika_chr.remove_acs(renpy.store.mas_acs_promisering)
             persistent._mas_acs_enable_promisering = False
 
+        #Change randchat
+        if store.persistent._mas_randchat_freq <= store.mas_randchat.LESS_OFTEN:
```
Comment (Booplicate) [inline Monika After Story/game/script-affection.rpy:494]:
```suggestion
        if store.persistent._mas_randchat_freq >= store.mas_randchat.LESS_OFTEN:
```

File: Monika After Story/game/script-affection.rpy [hunk 3]
```diff
@@ -505,6 +523,10 @@ init 15 python in mas_affection:
         # change quit message
         layout.QUIT_NO = mas_layout.QUIT_NO_UPSET
 
+        #Change randchat
+        if store.persistent._mas_randchat_freq <= store.mas_randchat.NORMAL:
```
Comment (Booplicate) [inline Monika After Story/game/script-affection.rpy:527]:
```suggestion
        if store.persistent._mas_randchat_freq >= store.mas_randchat.NORMAL:
```

File: Monika After Story/game/script-affection.rpy [hunk 4]
```diff
@@ -595,6 +617,10 @@ init 15 python in mas_affection:
         persistent._mas_monika_nickname = "Monika"
         m_name = persistent._mas_monika_nickname
 
+        #Change randchat
+        if store.persistent._mas_randchat_freq <= store.mas_randchat.OFTEN:
```
Comment (Booplicate) [inline Monika After Story/game/script-affection.rpy:621]:
```suggestion
        if store.persistent._mas_randchat_freq >= store.mas_randchat.OFTEN:
```

File: Monika After Story/game/script-affection.rpy [hunk 5]
```diff
@@ -632,6 +658,10 @@ init 15 python in mas_affection:
         # remove island event delayed actions
         store.mas_removeDelayedActions(1, 2)
 
+        #Change randchat
+        if store.persistent._mas_randchat_freq <= store.mas_randchat.VERY_OFTEN:
```
Comment (Booplicate) [inline Monika After Story/game/script-affection.rpy:662]:
```suggestion
        if store.persistent._mas_randchat_freq >= store.mas_randchat.VERY_OFTEN:
```

File: Monika After Story/game/definitions.rpy [hunk 6]
```diff
@@ -7346,15 +7355,24 @@ init 1 python in mas_randchat:
...
     rand_chat_waittime_left = 0
 
+    def reduceRandchatForAff(aff_level):
+        """
+        Reduces the randchat setting if we're too high for the current affection level
+        """
+        max_setting_for_level = store.mas_affection.RANDCHAT_RANGE_MAP[aff_level]
+
+        if store.persistent._mas_randchat_freq > max_setting_for_level:
+            adjustRandFreq(max_setting_for_level)
```
Comment (Booplicate) [inline Monika After Story/game/definitions.rpy:7365]: Actually, why don't we update this var too?
```suggestion
            adjustRandFreq(max_setting_for_level)
            store.mas_randchat_prev = max_setting_for_level
```

File: Monika After Story/game/updates.rpy [hunk 7]
```diff
@@ -378,6 +378,20 @@ label v0_11_4(version="v0_11_4"):
...
+            0: 6,
+            1: 5,
+            2: 4,
+            3: 3,
+            4: 2,
+            5: 1,
+            6: 0
+        }
+
+        persistent._mas_randchat_freq = OLD_NEW_RANDCHAT_MAP[persistent._mas_randchat_freq]
```
Comment (ThePotatoGuy) [inline Monika After Story/game/updates.rpy:392]:
```suggestion
        persistent._mas_randchat_freq = OLD_NEW_RANDCHAT_MAP.get(persistent._mas_randchat_freq, mas_randchat.NORMAL)
```
lets just make sure we don't crash at all in an update script.

**Pull request merged**